File: String/ReverseString.js
/**
 * 反转字符串
 * source: https://leetcode-cn.com/explore/featured/card/top-interview-questions-easy/5/strings/32/
 */



var reverseString = function(s) {
	let reverseS = '';
	for (var i = s.length - 1; i >= 0; i--) {
		reverseS += s.charAt(i);
	}    
	return reverseS;
};

var s="hello";
console.log(reverseString(s));

File: 初级算法/Swift/String/reverseString.swift
/**
 * 反转字符串
 * source: https://leetcode-cn.com/explore/interview/card/top-interview-questions-easy/5/strings/32/
 */

import Foundation

class Solution {
    func reverseString(_ s: inout [Character]) {
        for i in 0..<s.count/2 {
            let swapIndex = s.count-1-i
            let temp = s[i]
            s[i] = s[swapIndex]
            s[swapIndex] = temp
        }
    }
}
